c/cert/test/rules/CON30-C/test.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <threads.h>
+
+static tss_t k;
+
+void do_free(void *d) { free(d); }
+
+void maybe_free(void *d) {}
+
+void m1() {
+  tss_create(&k, free); // COMPLIANT
+  tss_delete(k);
+}
+
+void m2() {
+  tss_create(&k, do_free); // COMPLIANT
+  tss_delete(k);
+}
+
+void m3() {
+  tss_create(&k, maybe_free); // COMPLIANT
```

Comment:
// COMPLIANT[FALSE_NEGATIVE] ?

Reply:
even though its in the implementation notes? not essential, just consideration

Reply:
I reworked these test cases -- please have another look @knewbury01